networkx/generators/tests/test_community.py

```diff
@@ -133,6 +139,9 @@ def test_gaussian_random_partition_graph():
     pytest.raises(
         nx.NetworkXError, nx.gaussian_random_partition_graph, 100, 101, 10, 1, 0
     )
+    # Test when clusters are likely less than 1
+    G = nx.gaussian_random_partition_graph(10, 0.5, 0.5, 0.5, 0.5, seed=1)
+    assert len(G) == 10
```

Comment:
Hmm, in this test we aren't really testing the output of `gaussian_random_partition_graph`, as the length of nodes will always be input number of nodes. The other asserts here in `test_gaussian_random_partition_graph` also don't cover the functionality IMO. Here we are just testing the code is hitting all branches not if it's correct.

But we don't need to fix that in this PR, just writing some thoughts down :)

Reply:
Yeah I thought as much, but just figured I'd follow what was already there. From my understanding of the function, the edges are random, so a test on the size of the graph is not particularly helpful. Maybe a test with the probabilities set to 0 or 1 could help verify edge cases, but idk how to test a more general use case.

Reply:
Agreed - this would be a nice follow-up!